hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -414,6 +416,12 @@ protected DatanodeDetails chooseNodeFromNetworkTopology(
     Node pick = networkTopology.chooseRandom(
         anchor.getNetworkLocation(), excluded);
     DatanodeDetails pickedNode = (DatanodeDetails) pick;
+    if (pickedNode == null) {
+      LOG.debug("Pick node is null, excluded nodes {}, anchor {}.",
+          excluded, anchor);
+      throw new SCMException("Unable to find node based on Topology.",
+          SCMException.ResultCodes.FAILED_TO_FIND_SUITABLE_NODE);
+    }
```

Comment:
As I mentioned above, I feel it would be simpler if this just returns null rather than throwing an exception.

Reply:
Updated.